pkg/auditlogexport/credentials.go

```diff
@@ -0,0 +1,202 @@
+			} else {
+				credentialData["secret_access_key"] = existingCredentialData["secret_access_key"]
+			}
+		} else {
+			credentialData["access_key_id"] = ""
+			credentialData["secret_access_key"] = ""
+		}
+		credentialData["provider"] = string(types.StorageProviderS3)
+	case types.StorageProviderGCS:
+		if !config.UseWorkloadIdentity {
+			if config.GCSConfig.ServiceAccountJSON != "" {
+				credentialData["service_account_json"] = config.GCSConfig.ServiceAccountJSON
+			} else {
+				credentialData["service_account_json"] = existingCredentialData["service_account_json"]
+			}
+		} else {
+			credentialData["service_account_json"] = ""
+		}
+		credentialData["provider"] = string(types.StorageProviderGCS)
+	case types.StorageProviderAzureBlob:
+		if config.AzureConfig.StorageAccount != "" {
+			credentialData["storage_account"] = config.AzureConfig.StorageAccount
+		} else {
+			credentialData["storage_account"] = existingCredentialData["storage_account"]
+		}
+		if !config.UseWorkloadIdentity {
+			if config.AzureConfig.ClientID != "" {
+				credentialData["client_id"] = config.AzureConfig.ClientID
+			} else {
+				credentialData["client_id"] = existingCredentialData["client_id"]
+			}
+			if config.AzureConfig.TenantID != "" {
+				credentialData["tenant_id"] = config.AzureConfig.TenantID
+			} else {
+				credentialData["tenant_id"] = existingCredentialData["tenant_id"]
+			}
+			if config.AzureConfig.ClientSecret != "" {
+				credentialData["client_secret"] = config.AzureConfig.ClientSecret
+			} else {
+				credentialData["client_secret"] = existingCredentialData["client_secret"]
+			}
+		} else {
+			credentialData["client_id"] = ""
+			credentialData["tenant_id"] = ""
+			credentialData["client_secret"] = ""
+		}
+		credentialData["provider"] = string(types.StorageProviderAzureBlob)
+	case types.StorageProviderCustomS3:
+		if config.CustomS3Config.Endpoint != "" {
+			credentialData["endpoint"] = config.CustomS3Config.Endpoint
+		} else {
+			credentialData["endpoint"] = existingCredentialData["endpoint"]
+		}
+		if config.CustomS3Config.Region != "" {
+			credentialData["region"] = config.CustomS3Config.Region
+		} else {
+			credentialData["region"] = existingCredentialData["region"]
+		}
+		if config.CustomS3Config.AccessKeyID != "" {
+			credentialData["access_key_id"] = config.CustomS3Config.AccessKeyID
+		} else {
+			credentialData["access_key_id"] = existingCredentialData["access_key_id"]
+		}
+		if config.CustomS3Config.SecretAccessKey != "" {
+			credentialData["secret_access_key"] = config.CustomS3Config.SecretAccessKey
+		} else {
+			credentialData["secret_access_key"] = existingCredentialData["secret_access_key"]
+		}
+		credentialData["provider"] = string(types.StorageProviderCustomS3)
+	}
+
+	if err := g.gptClient.DeleteCredential(ctx, storeageCredentialsContext, storageCredentialsName); err != nil && !errors.As(err, &gptscript.ErrNotFound{}) {
+		return err
+	}
+
```

Comment:
This can actually be removed. We thought that you had to delete the credential before creating it, but you can just call create and it will work.